redis/db/redisbp/monitored_client.go

```diff
@@ -72,6 +83,75 @@ func (cc *ClusterClient) Wait(ctx context.Context, args WaitArgs) (replicas int6
 	return
 }
 
+var (
+	gaugeLabels = []string{
+		nameLabel,
+	}
+
+	totalConnectionsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(promNamespace, subsystemPool, "connections"),
+		"Number of total connections in this redisbp pool",
+		gaugeLabels,
+		nil,
+	)
+
+	idleConnectionsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(promNamespace, subsystemPool, "idle_connections"),
+		"Number of idle connections in this redisbp pool",
+		gaugeLabels,
+		nil,
+	)
+
+	staleConnectionsDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(promNamespace, subsystemPool, "stale_connections"),
+		"Number of stale connections in this redisbp pool",
+		gaugeLabels,
+		nil,
+	)
+)
+
+// Exporter provides an interface for Prometheus merics.
+type Exporter struct {
+	client PoolStatser
+	name   string
+}
+
+func NewExporter(client PoolStatser, name string) *Exporter {
+	return &Exporter{
+		client: client,
+		name:   name,
+	}
+}
```

Comment:
I don't think we need to export either `Exporter` or `NewExporter`? Or if you expect people to construct the client themselves without using `NewMonitoredClient`, then `NewExporter` might be useful but `Exporter` still does not need to be exported (or just remove `NewExporter` and keep `Exporter`, as you didn't actually do anything extra inside `NewExporter`).

Reply:
Sure, those can also be private. I will make this change.